### ISSUE TITLE
FEATURE - Integrate with Bookmarks API to add previously bookmarked tweets

### ISSUE THREAD

**Issue by jakedgrant** [mikaelacaron/brain-marks#123]
**Is your feature request related to a problem? Please describe.**
I have previously bookmarked tweets that I would like to add to Brain Marks but adding these one-by-one can be time consuming.

**Describe the solution you'd like**
Using the new twitter [bookmarks api](https://developer.twitter.com/en/docs/twitter-api/tweets/bookmarks/introduction), allow the user to organize these tweets into categories. Import all bookmarks to an unassigned category at first, then allow the user to assign categories later.

**Describe alternatives you've considered**
- Copy and pasting previously bookmarked tweets is time consuming.
- Share sheet extension could be another alternative (#57)

**Additional context**
The bookmarks api also allows for removing bookmarks from a tweet. Brain Marks could use the DELETE command to manage these tweets as well.

Might want to consider if existing tweets in Brain Marks should POST to twitter as bookmarks. This would have Brain Marks synced with a user's bookmarked tweets.

**Comment by mikaelacaron** [comment]
Oh interesting! I didn't see that Twitter added a new API!
I definitely want to integrate #57 , but this new API looks really nice too!

To use the bookmarks API though, it would mean users need to log into their Twitter accounts right?

**Comment by jakedgrant** [comment]
Yeah, it would require generating a user access token with OAuth 2.0.

**Comment by mikaelacaron** [comment]
Keeping the bookmarks in sync, how would that work? If you added a new bookmark on Twitter, would it be possible to trigger that to push to Brain Marks? Or on app launch Brain Marks would always look for any new tweets?

I like this feature! I think it needs to be broken up some more, cause it's pretty big!

**Comment by mikaelacaron** [comment]
Putting this on hold for now, during Hacktoberfest 2022, but this would be great to add in the future!
